src/pykeen/nn/node_piece.py

```diff
@@ -565,6 +566,127 @@ def resolve_aggregation(
     return aggregation
 
 
+class TokenizationRepresentationModule(RepresentationModule):
+    """A module holding the result of tokenization."""
+
+    #: the token ID of the padding token
+    vocabulary_size: int
+
+    #: the token representations
+    vocabulary: RepresentationModule
+
+    #: the assigned tokens for each entity
+    assignment: torch.LongTensor
+
+    def __init__(
+        self,
+        assignment: torch.LongTensor,
+        token_representation: HintOrType[RepresentationModule] = None,
+        token_representation_kwargs: OptionalKwargs = None,
+    ) -> None:
+        """
+        Initialize the tokenization.
+
+        :param assignment: shape: `(n, num_chosen_tokens)`
+            the token assignment.
+        :param token_representation: shape: `(num_total_tokens, *shape)`
+            the token representations
+        :param token_representation_kwargs:
+            additional keyword-based parameters
+        """
+        # needs to be lazily imported to avoid cyclic imports
+        from . import representation_resolver
+
+        # fill padding (nn.Embedding cannot deal with negative indices)
+        padding = assignment < 0
+        assignment[padding] = self.vocabulary_size = vocabulary_size = assignment.max().item() + 1
```

Comment:
triple assignment?!

Reply:
seems redundant - get rid of `vocabulary size` and on line 610 you can use self.vocabulary_size... unless this does some weird magic

Reply:
-> 992c8b0

Reply:
I think this was a refactoring artifact from having a separate `padding_idx` attribute.